one and done #41 - test task details

dabiid opened the pull request:
This has all the changes based on the discussion in pr #49. I squashed the commits into one since I'm starting a new pr. Most recent changes are to select team link by href contains 'team' and to change team_details page to use the page header (and then an assert name in header in the test). Also removed page title from the team_details page because there isn't a clean way to get the team name right now.

bobsilverberg commented:
This looks great, thanks @dabiid! I'm going to merge it now.

Thanks for this excellent contribution. If you don't mind my asking, I was wondering how you found this issue to work on? We're always interested in knowing from whence our contributors are coming.

bobsilverberg commented on tests/test_task_details.py:
```diff
@@ -0,0 +1,27 @@
+
+class TestTaskDetails():
+
+    def test_that_person_can_view_task_details(self, base_url, selenium):
```
Oops, I just realized that there is one thing missing here. Because this test does not change any data it can be marked as `nondestructive`. If we do not do that then we won't be able to run it against production. To mark it as `nondestructive`, you simply add the pytest nondestructive mark. See https://github.com/mozilla/oneanddone-tests/blob/master/tests/test_homepage.py#L12 for an example.

Please let me know once that is done, and my apologies for missing that the first time around.

dabiid commented:
I found it on the one and done page. I filtered web QA by automated tasks and it came up. 

https://oneanddone.mozilla.org/tasks/149/

It took me a bit to figure out how github fit into the mix even though it's pretty clear in the explanation/verification steps what to do (at least in retrospect with my new knowledge of how github works).

dabiid commented:
Updated with the nondestructive decorator. I squashed the commit since it was a small change and should be the last one - let me know if I should just leave it in the future and squash after.

bobsilverberg commented:
Perfect, thanks @dabiid! We do generally prefer that commits not be squashed until the whole review process is completed, because it makes subsequent reviews much easier, but in this case, with a one-line change, it's fine. In the future though it's probably best to err on the side of caution and not squash commits until the review is signed off.

Thanks again for all of your efforts with this - it's great work. I will go and look at your other pull request now.
